### PR TITLE
feat(axe-core-4.10.2): update to use axe-core 4.10.2

### PR DESCRIPTION
**Pull request by JeevaniChinthala**
#### Details

This PR upgrades axe-core to 4.10.2 according to the instructions in CONTRIBUTING.md.

##### Motivation

feature work 🚀
#### Pull request checklist

- [x] PR title respects [Conventional Commits](https://www.conventionalcommits.org) (starts with `fix:`, `feat:`, etc, and is suitable for user-facing release notes)
- [x] PR contains no breaking changes, **OR** description of both PR **and final merge commit** starts with `BREAKING CHANGE:`
- [x] Verify PR title and final merge commit contain space after `:` for example `feat(feature): feature title` otherwise it will not be considered by semantic-release for release.
- [ ] (if applicable) Addresses issue: #0000
- [ ] Added relevant unit tests for your changes
- [x] Ran `yarn precheckin`
- [x] Verified code coverage for the changes made

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -48,81 +48,93 @@ describe('public convertAxeToSarif API', () => {
...
+        ${'basic-axe-v4.8.4.reporter-v2.json'}         | ${'basic-axe-v4.8.4.sarif'}
+        ${'w3citylights-axe-v4.8.4.reporter-v1.json'}  | ${'w3citylights-axe-v4.8.4.sarif'}
+        ${'w3citylights-axe-v4.8.4.reporter-v2.json'}  | ${'w3citylights-axe-v4.8.4.sarif'}
+        ${'basic-axe-v4.9.1.reporter-v1.json'}         | ${'basic-axe-v4.9.1.sarif'}
+        ${'basic-axe-v4.9.1.reporter-v2.json'}         | ${'basic-axe-v4.9.1.sarif'}
+        ${'w3citylights-axe-v4.9.1.reporter-v1.json'}  | ${'w3citylights-axe-v4.9.1.sarif'}
+        ${'w3citylights-axe-v4.9.1.reporter-v2.json'}  | ${'w3citylights-axe-v4.9.1.sarif'}
+        ${'basic-axe-v4.10.0.reporter-v1.json'}        | ${'basic-axe-v4.10.0.sarif'}
+        ${'basic-axe-v4.10.0.reporter-v2.json'}        | ${'basic-axe-v4.10.0.sarif'}
+        ${'w3citylights-axe-v4.10.0.reporter-v1.json'} | ${'w3citylights-axe-v4.10.0.sarif'}
```
Comment (madalynrose) [inline src/index.test.ts:128]: do we need the versions that we're not switching to? I'd expect for this update to only add lines with `4.10.2`

**Review by madalynrose — CHANGES_REQUESTED** [review]
This generally looks as expected except I think the files and updates for versions `4.10.0` and `4.10.1` are superfluous. I think we only need new files and tests for `4.10.2`.

File: src/index.test.ts (hunk 1 above)
Comment (JeevaniChinthala) [reply src/index.test.ts]: Updated files for 4.10.2 version

**Review by JGibson2019 — APPROVED** [review]

**Review by madalynrose — APPROVED** [review]

**Pull request merged**